core/src/main/java/org/keycloak/jose/jwk/JWKBuilder.java

```diff
@@ -17,14 +17,48 @@
 
 package org.keycloak.jose.jwk;
 
-import org.keycloak.crypto.KeyUse;
+import static org.keycloak.jose.jwk.JWKUtil.toIntegerBytes;
 
 import java.security.Key;
+import java.security.PublicKey;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Collections;
+import java.util.List;
+
+import org.keycloak.common.util.Base64Url;
+import org.keycloak.common.util.KeyUtils;
+import org.keycloak.common.util.PemUtils;
+import org.keycloak.crypto.Algorithm;
+import org.keycloak.crypto.KeyType;
+import org.keycloak.crypto.KeyUse;
 
 /**
  * @author <a href="mailto:sthorger@redhat.com">Stian Thorgersen</a>
  */
-public class JWKBuilder extends AbstractJWKBuilder {
+public class JWKBuilder {
+
+    // internal util class only loaded for jdk versions with EdEC support
+    protected static final EdECUtils EdEC_UTILS;
+
+    static {
+        EdECUtils tmp;
+        try {
+            // check if the impl class for EdEC can be loaded in the runtime
+            tmp = (EdECUtils) Class.forName("org.keycloak.jose.jwk.EdECUtilsImpl")
+                    .getDeclaredConstructor().newInstance();
+        } catch(Throwable e) {
+            tmp = null;
```

Comment:
My personal preference is possibly to use something like:
```
tmp = new EdECUtils() {
    
    @Override
    public JWK okp(Key key, KeyUse keyUse) {
        throw new UnsupportedOperationException("EdDSA algorithms not supported in this JDK version");
    }

   ...
}
```

instead of null checks. But that is maybe just me and it does not matter much IMO (especially since the field is protected anyway). So feel free to keep it as it is unless you strongly prefer to update or there are some more comments...

Reply:
Yep, we can do that. But then we need a `isSupported` method to follow the current behavior. Give me 5 min and I'll send a PR.

Reply:
Done! (I have used a separated class file for the unsupported version.)